src/app/clusters/fault-injection-server/fault-injection-server.cpp

```diff
@@ -93,4 +93,66 @@ bool emberAfFaultInjectionClusterFailAtFaultCallback(CommandHandler * commandObj
     return true;
 }
 
+bool emberAfFaultInjectionClusterFailRandomlyAtFaultCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                             const Commands::FailRandomlyAtFault::DecodableType & commandData)
+{
+    if (commandPath.mClusterId != Clusters::FaultInjection::Id)
+    {
+        // We shouldn't have been called at all.
+        commandObj->AddStatus(commandPath, Status::UnsupportedCluster);
+        return true;
+    }
+
+    if (commandData.percentage > 100)
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+#if CHIP_WITH_NLFAULTINJECTION
+    Status returnStatus                             = Status::Success;
+    nl::FaultInjection::Manager * faultInjectionMgr = nullptr;
+
+    switch (commandData.type)
```

Comment:
would you mind pulling this section out into a separate function so it's not duplicated? Would make it easier to add new types if necessary.

Reply:
Acked